src/data/sum/interval.lean

```diff
@@ -0,0 +1,165 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.sum.order
+import order.locally_finite
+
+/-!
+# Finite intervals in a disjoint union
+
+This file provides the `locally_finite_order` instance for the disjoint sum of two orders and
+calculates the cardinality of its finite intervals.
```

Comment:
Does it?

Reply:
Meh, not really. This will be true for the lexicographic sum.